pymc/distributions/mixture.py

```diff
@@ -176,6 +164,17 @@ def dist(cls, w, comp_dists, **kwargs):
                 UserWarning,
             )
 
+        if len(comp_dists) > 1:
+            all_continuous = all([comp_dist.dtype in continuous_types for comp_dist in comp_dists])
+            all_discrete = all([comp_dist.dtype in discrete_types for comp_dist in comp_dists])
```

Comment:
Micro-optimization, we don't need to wrap them in a list if we only care about the boolean. Also if you put them in an `or` `if` statement it will not need to evaluate the second expression if the first is already `True`

```suggestion
            all_continuous = all(comp_dist.dtype in continuous_types for comp_dist in comp_dists)
            all_discrete = all(comp_dist.dtype in discrete_types for comp_dist in comp_dists)
```

Reply:
I agree with the list wrapping suggestion. Fixed!

However, would using `if`/`or` for cases where we don't have to evaluate the second expression be very beneficial in terms of code optimization? My first instinct is that defining it this way is better for readability purposes, but I don't know what would the best choice here

Reply:
In general, there is no reason to compute something we don't need afterwards. In this case we only need an if / else check so we should not compute everything. I think this is still pretty readable:

```python
 if not (
  all(comp_dist.dtype in continuous_types for comp_dist in comp_dists)
  or all(comp_dist.dtype in discrete_types for comp_dist in comp_dists)
):
  raise ValueError(...)